src/bio/bio_config.c

```diff
@@ -984,45 +946,45 @@ int
 bio_read_auto_faulty_criteria(const char *nvme_conf, bool *enable, uint32_t *max_io_errs,
 			      uint32_t *max_csum_errs)
 {
-	struct config_entry     cfg                  = {};
 	struct auto_faulty_info auto_faulty_criteria = {};
 	int                     rc;
 
-	rc = decode_daos_data(nvme_conf, NVME_CONF_SET_AUTO_FAULTY, &cfg);
-	if (rc != 0)
-		goto out;
-
-	rc = spdk_json_decode_object(cfg.params, auto_faulty_decoders,
-				     SPDK_COUNTOF(auto_faulty_decoders), &auto_faulty_criteria);
-	if (rc < 0) {
-		D_ERROR("Failed to decode '%s' entry: %s)\n", NVME_CONF_SET_AUTO_FAULTY,
-			spdk_strerror(-rc));
-		D_GOTO(out, rc = -DER_INVAL);
+	D_ASSERT(enable != NULL);
+	D_ASSERT(max_io_errs != NULL);
+	D_ASSERT(max_csum_errs != NULL);
+
+	rc = decode_daos_object(nvme_conf, NVME_CONF_SET_AUTO_FAULTY, auto_faulty_decoders,
+				SPDK_COUNTOF(auto_faulty_decoders), &auto_faulty_criteria);
+	if (rc != 0) {
+		if (rc == JSON_NOT_FOUND) {
+			rc = 0;
+			*enable        = false;
+			*max_io_errs   = UINT32_MAX;
+			*max_csum_errs = UINT32_MAX;
```

Comment:
FYI clang-format is complaining because this is not aligned

Reply:
keeping as is so we have the same change on master and 2.6 branches